### PR TITLE
build(documentation-website): add cypress test

### LINKED ISSUE #912: Cookie-consent form cypress tests
## Context

Cookie consent form is added and needs to be tested.

## Test

Test the form, the functionality of the inputs and buttons.

## Result

The form should 

- exist
- have functional inputs for motomo and youtube tracking
- have function buttons to accept, customize, and decline cookies

### PR DESCRIPTION
**Pull request by jauntyjocularjay**
added cypress test for cookie-form on /home
closes #912

# The Pull Request is ready

- [ ] fixes #
- [ ] all actions are passing
- [ ] only fixes a single issue

## Documentation-Website

- [x] mobile view is usable
- [x] desktop view is usable
- [x] no a-tags are used directly (NavLink, MailLink, ExternalLink instead)
- [x] all new texts are added to the translation files (at least the english one)
- [x] tests have been added (if required)
- [x] shared code has been extracted in a different file

### CODE REVIEW

File: documentation-website/cypress/e2e/home.cy.ts [hunk 1]
```diff
@@ -36,5 +38,65 @@
...
+    cy.get('form.cookie-consent', flakeTimeout,)
+      .as('form',);
+    cy.get('ul#consent-choices', flakeTimeout,)
+      .as('choices',);
+    cy.get('@choices',)
+      .should('exist',);
+    cy.get('@choices',)
+      .find('li.service',)
+      .then(
+        li => {
```
[the remark this answers is not in the document]
Comment (jauntyjocularjay) [reply documentation-website/cypress/e2e/home.cy.ts]: No changes to make

File: cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export default {
```
Comment (Idrinth) [inline cypress.config.ts]: this is not meant to be here

File: cypress/fixtures/example.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (Idrinth) [inline cypress/fixtures/example.json]: this is not meant to be here

File: cypress/support/commands.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
```
Comment (Idrinth) [inline cypress/support/commands.ts]: this is not meant to be here

File: cypress/support/e2e.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// ***********************************************************
```
Comment (Idrinth) [inline cypress/support/e2e.ts]: this is not meant to be here

File: gitCommit.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+read -rp "Enter the task summary: " summary
+read -rp "Enter the task description: " description
+read -rp "What issue number does this address? " issue
+
+git add .
+git commit -m "$task($scope): $summary \n $description \n closes #$issue"
+git pull
+git push
+
+git checkout the-one
```
Comment (Idrinth) [inline gitCommit.sh]: this breaks the logic, because you will be on the-one afterwards - are you sure you didn't just want to fetch the-one?

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: cypress.config.ts (hunk 2 above)
Comment (jauntyjocularjay) [reply cypress.config.ts]: Why do these keep getting put into the root? I didn't open Cypress in the root.

File: gitCommit.sh (hunk 6 above)
Comment (jauntyjocularjay) [reply gitCommit.sh]: In what what does this break the logic?

All of these changes affect the developer's fork, not the original. So what I am doing with this is apply changes from the local working branch (`dev-cypress` in my case) to the local `the-one` from the developer's fork. I added a pull after changing to the `fork: the-one`. That can potentially break the logic.

File: cypress.config.ts (hunk 2 above)
Comment (Idrinth) [reply cypress.config.ts]: weird :(

File: gitCommit.sh (hunk 6 above)
Comment (jauntyjocularjay) [reply gitCommit.sh]: The script works, I have verified it several times. Barring an explanation of how the script is broken, it is as it is intended to be.

File: gitCommitMain.sh [hunk 7]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# Commit changes form for easy commits
+
+read -rp "Enter your working branch name: " branch
```
[the remark this answers is not in the document]
Comment (jauntyjocularjay) [reply gitCommitMain.sh]: outdated

File: gitCommitDev.sh [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+# Commit changes form for easy commits
+
+read -rp "Enter your task type: " task
+read -rp "Enter the task scope: " scope
+read -rp "Enter the task summary: " summary
+read -rp "Enter the task description: " description
+read -rp "What issue number does this address? " issue
+
+git add .
+git commit -m "$task($scope): $summary \n $description \n closes #$issue"
```
[the remark this answers is not in the document]
Comment (jauntyjocularjay) [reply gitCommitDev.sh:37]: not echoing

File: gitCommitMain.sh [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+read -rp "Enter your task type: " task
+read -rp "Enter the task scope: " scope
+read -rp "Enter the task summary: " summary
+read -rp "Enter the task description: " description
+read -rp "What issue number does this address? " issue
+
+git pull
+git add .
+git commit -m "$task($scope): $summary \n $description \n closes #$issue"
```
[the remark this answers is not in the document]
Comment (jauntyjocularjay) [reply gitCommitMain.sh]: Not echoing

File: gitCommit.sh [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+read -rp "Enter your task type: " task
+read -rp "Enter the task scope: " scope
+read -rp "Enter the task summary: " summary
+read -rp "Enter the task description: " description
+read -rp "What issue number does this address? " issue
+
+git add .
+git commit -m "$task($scope): $summary \n $description \n closes #$issue"
+git pull
+git push
```
[the remark this answers is not in the document]
Comment (jauntyjocularjay) [reply gitCommit.sh]: outdated

File: gitCommitOne.sh [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+git pull
+git add .
+git commit -m "$task($scope): $summary \n $description \n closes #$issue"
+git push
+
+git checkout the-one
+git pull
+git merge $branch
+git push
```
[the remark this answers is not in the document]
Comment (jauntyjocularjay) [reply gitCommitOne.sh]: No, we have a very specific name for our master branch.

File: gitCommitOne.sh [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+read -rp "Enter the task summary: " summary
+echo ""
+read -rp "Enter the task description: " description
+echo ""
+read -rp "What issue number does this address? " issue
+echo ""
+
+git pull
+git add .
+git commit -m "$task($scope): $summary \n $description \n closes #$issue"
```
[the remark this answers is not in the document]
Comment (jauntyjocularjay) [reply gitCommitOne.sh]: Mine is more readable

**Review by Idrinth — DISMISSED** [review]

**Pull request merged**